hs_core/templates/pages/genericresource.html

```diff
@@ -468,363 +490,365 @@
                     {% else %}
                         <strong class="label-private">&nbspPrivate</strong>
                     {% endif %}
+                    &{% if cm.raccess.shareable %}<strong style="color:#5cb85c">&nbspShareable</strong>{% else %}<strong style="color:#d9534f">&nbspNot Shareable</strong>{% endif %}
                 {% endif %}
             </div>
 
-            {% if is_owner_user %}
-                <div><p>You are the owner of this resource.</p></div>
-            {% elif is_edit_user %}
-                <p>You have been given specific permission to edit this resource.</p>
-            {% elif is_view_user %}
-                <p>You have been given specific permission to view this resource.</p>
-            {% endif %}
+            <div id="user-permission-msg">
+                {% if is_owner_user %}
+                    <div class="clear-fix"><i>You are the owner of this resource.</i></div>
+                {% elif is_edit_user %}
+                     <div class="clear-fix"><i>You have been given specific permission to edit this resource.</i></div>
+                {% elif is_view_user %}
+                     <div class="clear-fix"><i>You have been given specific permission to view this resource.</i></div>
+                {% endif %}
+            </div>
+            {% if is_owner_user or cm.raccess.shareable and is_view_user or cm.raccess.shareable and is_edit_user %}
```

Comment:
I suggest to add parenthesis in this {% if ..%} condition to make the logic clearer - when "and" and "or" are mixed in a conditional statement, the logic is not clear without parenthesis.